src/Features/Core/Portable/NavigateTo/NavigateToItem.cs

```diff
@@ -0,0 +1,272 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Runtime.Serialization;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Navigation;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.NavigateTo
+{
+    /// <summary>
+    /// Data about a navigate to match.  Only intended for use by C# and VB.  Carries enough rich information to
+    /// rehydrate everything needed quickly on either the host or remote side.
+    /// </summary>
+    [DataContract]
+    internal readonly struct RoslynNavigateToItem
+    {
+        [DataMember(Order = 0)]
+        public readonly bool IsStale;
```

Comment:
stale-ness is a concept that will be in the next pr.  currently all results are non-stale.  however, in the next PR it will be ok for a stale-result to be returned.  the host can then filter it out if that doc doesn't exist in teh current solution.  Navigation will also have to be resilient to the span possibly being out of date.